src/Command/DumpConfigCommand.php

```diff
@@ -0,0 +1,83 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Command;
+
+use Nette\Neon\Neon;
+use PHPStan\ShouldNotHappenException;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use function is_array;
+use function is_string;
+
+class DumpConfigCommand extends Command
+{
+
+	private const NAME = 'dump-config';
+
+	/**
+	 * @param string[] $composerAutoloaderProjectPaths
+	 */
+	public function __construct(
+		private array $composerAutoloaderProjectPaths,
+	)
+	{
+		parent::__construct();
+	}
+
+	protected function configure(): void
+	{
+		$this->setName(self::NAME)
+			->setDescription('Dumps config')
+			->setDefinition([
+				new InputArgument('paths', InputArgument::OPTIONAL | InputArgument::IS_ARRAY, 'Paths with source code to run analysis on'),
+				new InputOption('configuration', 'c', InputOption::VALUE_REQUIRED, 'Path to project configuration file'),
+				new InputOption(AnalyseCommand::OPTION_LEVEL, 'l', InputOption::VALUE_REQUIRED, 'Level of rule options - the higher the stricter'),
+				new InputOption('autoload-file', 'a', InputOption::VALUE_REQUIRED, 'Project\'s additional autoload file path'),
```

Comment:
I think we can remove paths, but add memory-limit here.

Basically the arguments/options should mirror ClearResultCacheCommand, and add level because that influences the loaded file :)